netdiff/utils.py

```diff
@@ -134,35 +179,26 @@ def _netjson_networkgraph(
         raise NetJsonError('version cannot be None except when protocol is "static"')
     # prepare nodes
     node_list = []
-    for node in nodes:
-        netjson_node = OrderedDict({'id': node[0]})
+    for ip, data in nodes:
+        netjson_node = OrderedDict({'id': ip})
         # must copy properties dict to avoid modifying data
-        properties = node[1].copy()
-        local_addresses = properties.pop('local_addresses', None)
-        label = properties.pop('label', None)
-        # append local_addresses only if not empty
-        if local_addresses:
-            netjson_node['local_addresses'] = local_addresses
-        # append properties only if not empty
-        if properties:
-            netjson_node['properties'] = properties
-        # append label only if not empty
-        if label:
-            netjson_node['label'] = label
+        properties = data.copy()
+        netjson_node['label'] = properties.pop('label', '') or ''
+        netjson_node['local_addresses'] = properties.pop('local_addresses', [])
+        netjson_node['properties'] = properties or {}
         node_list.append(netjson_node)
+    node_list.sort(key=lambda d: d['id'])
     # prepare links
     link_list = []
-    for link in links:
+    for src, dst, props in links:
         # must copy properties dict to avoid modifying data
-        properties = link[2].copy()
-        cost = properties.pop('weight')
-        netjson_link = OrderedDict(
-            (('source', link[0]), ('target', link[1]), ('cost', cost))
-        )
-        # append properties only if not empty
-        if properties:
-            netjson_link['properties'] = properties
+        properties = props.copy()
+        netjson_link = OrderedDict((('source', src), ('target', dst)))
+        netjson_link['cost'] = properties.pop('weight')
+        netjson_link['cost_text'] = properties.pop('cost_text', '')
+        netjson_link['properties'] = properties or {}
```

Comment:
also here, the `or {}` looks redundant, bacause above you do: `props.copy()`, and `copy()` is a method of dict, so it `props` is not a dict that line above would raise an exception before getting to de `or {}`

Reply:
This is correct, I modified it previously and forgot to update it